### PR TITLE
Don't serialize apply_soft_confirmation_txs txs

### LINKED ISSUE #1452: StfBlueprintTrait::apply_soft_confirmation_txs - accept Vec<Transaction> instead of Vec<Vec<u8>>
So we don't need to borsh::from_slice for Transaction<C: Context>.

### PR DESCRIPTION
**Pull request by kpp**
# Description

Pass them unborshed into zk context.

## Linked Issues
- Fixes https://github.com/chainwayxyz/citrea/issues/1452

### CODE REVIEW

File: crates/sovereign-sdk/module-system/sov-modules-api/src/transaction.rs [hunk 1]
```diff
@@ -71,6 +72,16 @@ impl<C: Context> Transaction<C> {
     }
 }
 
+impl<C: Context> TransactionTrait for Transaction<C> {
```
Comment (kpp) [inline crates/sovereign-sdk/module-system/sov-modules-api/src/transaction.rs]: I am not sure if I can calc hash this way or I need to add a hash field to Transaction struct

File: crates/batch-prover/src/da_block_handler.rs [hunk 2]
```diff
@@ -66,9 +66,10 @@ where
     pending_l1_blocks: VecDeque<<Da as DaService>::FilteredBlock>,
     _state_root: PhantomData<StateRoot>,
     _witness: PhantomData<Witness>,
+    _tx: PhantomData<fn() -> Tx>,
```
Comment (rakanalh) [inline crates/batch-prover/src/da_block_handler.rs]:
```suggestion
    _tx: PhantomData<Tx>,
```
Why not this?

File: crates/batch-prover/src/da_block_handler.rs [hunk 3]
```diff
@@ -82,6 +83,7 @@ where
         + AsRef<[u8]>
         + Debug,
     Witness: Default + BorshDeserialize + BorshSerialize + Serialize + DeserializeOwned,
+    Tx: Clone + BorshDeserialize + BorshSerialize + 'static,
```
Comment (rakanalh) [inline crates/batch-prover/src/da_block_handler.rs]: You're using `'static` anyway so i don't see the point of `fn() ->` part above.

File: crates/sovereign-sdk/rollup-interface/src/state_machine/stf.rs [hunk 4]
```diff
@@ -182,6 +188,14 @@ pub struct SoftConfirmationResult<S, Cs, T, W, Da: DaSpec> {
 ///  - batch: Set of transactions grouped together, or block on L2
 ///  - blob: Non serialised batch or anything else that can be posted on DA layer, like attestation or proof.
 pub trait StateTransitionFunction<Da: DaSpec> {
+    /// The type of rollup transaction
+    type Transaction: TransactionTrait
```
Comment (rakanalh) [inline crates/sovereign-sdk/rollup-interface/src/state_machine/stf.rs]:
```suggestion
    type Transaction: TransactionDigest
```

File: crates/sovereign-sdk/rollup-interface/src/state_machine/soft_confirmation.rs [hunk 5]
```diff
@@ -13,18 +13,19 @@ use serde::{Deserialize, Serialize};
...
 /// Contains raw transactions and information about the soft confirmation block
 #[derive(Debug, PartialEq, BorshSerialize)]
-pub struct UnsignedSoftConfirmation<'txs> {
+pub struct UnsignedSoftConfirmation<'txs, Tx> {
     l2_height: u64,
     da_slot_height: u64,
     da_slot_hash: [u8; 32],
     da_slot_txs_commitment: [u8; 32],
     txs: &'txs [Vec<u8>],
+    txs_new: &'txs [Tx],
```
Comment (rakanalh) [inline crates/sovereign-sdk/rollup-interface/src/state_machine/soft_confirmation.rs]: We can probably find a better name for this.

We can name `txs` as `serialized_txs` and leave `txs` as the unserialized ones.

Comment (rakanalh) [reply crates/sovereign-sdk/rollup-interface/src/state_machine/soft_confirmation.rs]: Or `txs` becomes `blobs` and `txs_new` becomes `txs`

File: crates/batch-prover/src/da_block_handler.rs (hunk 2 above)
Comment (kpp) [reply crates/batch-prover/src/da_block_handler.rs:69]: https://www.reddit.com/r/rust/comments/1cktu0g/comment/l2pwii0/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

File: crates/sovereign-sdk/rollup-interface/src/state_machine/soft_confirmation.rs (hunk 5 above)
Comment (kpp) [reply crates/sovereign-sdk/rollup-interface/src/state_machine/soft_confirmation.rs]: I mean that's a temporary name until we get rid of Fork1

File: crates/sovereign-sdk/module-system/sov-modules-api/src/transaction.rs (hunk 1 above)
Comment (kpp) [reply crates/sovereign-sdk/module-system/sov-modules-api/src/transaction.rs]: @eyusufatik ^

Comment (eyusufatik) [reply crates/sovereign-sdk/module-system/sov-modules-api/src/transaction.rs]: depends on how many times we invoke the hash func.

if only once, we don't need it in the struct. else, it's better to have some form of calculate and then reuse the calculated value.

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**